README.md

```diff
@@ -10,327 +10,66 @@ New projects should follow the Android Gradle project structure that is defined
 
 It's recommended to structure in the following way:
 
-### App
+    mainpackage
+     * homescreen 			// 1.1.1. Activity level
+	    * category 		// 1.1.2 Feature of the activity
+            * di		// 1.1.3 Dependency Injection
+            * presenter	// 1.1.4 Presentation layer
+            * repository	// 1.1.5 Repository layer
+            * navigator	// 1.1.6 Navigator layer
+        * search
+            * di 
+            * presenter
+            * repository
+            * navigator
 
-* network
-	* picasso
-	* dashboard 	
-	* matchfeed
-* model 
-* injection
-	* modules
-	* component
-	* qualifiers	 	
-* ui
-	* dashboard
-	* notifications 	
-	* matchfeed
-* persistence
-	* dashboard
-	* matchfeed	
-* analytics
-* ads
-* utils
-
-#### 1.1.1 Network
+#### 1.1.1 Activity level 
 
-Contains all the network related files (Http services, deserializers, image loading...). Try to group network files for different parts of the app in different subpackages.
+Activity providing a context for multiple features
 
-#### 1.1.2 Model
+#### 1.1.2 Feature level
 
-Contains all the model related files.
+Feature level encapsulates all packages and classes required for delivering specific feature.
 
 #### 1.1.3 Injection
 
-We use depency injection usually using [Dagger 2](http://google.github.io/dagger/). This package will contain all the dependency injection related files (modules, components, qualifiers, scopes...).
-
-#### 1.1.4 Ui
+Injection package will contain all components, modules, qualifiers and scopes used by this feature. If this feature requires more elaborate injection it is worth to split the package further by role 
 
-Contains all ui related files (Views, adapters...). Try to group ui files for different parts of the app into different subpackages.
-
-#### 1.1.5 Persistence
+	* components
+	* modules
+	* scopes
+	* qualifiers
 
-Contains all the persistence related files (Shared preferences, sqlite, files storage...). Try to group persistence files for different parts of the app into different subpackages.
+#### 1.1.4 Presenter
 
-#### 1.1.6 Analytics
+Presentation layer will contain interface and implementation(s) of view logic
 
-Contains all the analytics related files
+#### 1.1.5 Repository
 
-#### 1.1.7 Ads
+Repository package will encapsulate all interfaces and logic around requesting data from external and internal sources. It may contain packages e.g.
 
-Contains all the adverts related files
+	* parsers
+	* models
+	* handlers
 
-#### 1.1.8 Utils
+#### 1.1.6 Navigator
 
-Contains all the utility related files, usually they are files containing only static methods that helps to do some specific task that it's repeated all over the app and doesn't really belong to any other part of the app (TimeUtils, FontUtils, ViewUtils...) 
+Navigator interface defines all navigational operations that can be performed on Fragments/Activity inside this Activity. e.g.
 
+    public interface HomeSearchNavigator {
```

Comment:
I've seen "Navigator" objects implemented at TAB and in my personal opinion they end up always being a mess and implemented by the activities, which specially it we are following the MVC pattern this logic is part of the controller. The controller could delegate this logic to a "navigator" but generally if the app is not really complex it's not even worth it, plus I recommend the intent to be created as a static method in the activity you want to launch, that way is really easy to know entry points to that activity/fragment